processes_notwindows.go

```diff
@@ -107,6 +112,23 @@ func processesFromProc() ([]ProcStat, error) {
 			stat.Cmdline = strings.Replace(string(bytes.TrimRight(cmdline, "\x00")), "\x00", " ", -1)
 		}
 
+		cgroupFilepath := getHostProc() + "/" + pidString + "/cgroup"
+		cgroup, err := readProcFile(cgroupFilepath)
+		if err != nil && err != errorProcessTerminated {
+			log.Errorf("[PROC] failed to read cgroup(%s): %s", cgroupFilepath, err.Error())
```

Comment:
can we call `continue` here and then un-indent  the `else` branch?

Reply:
no, we can't, it still needs to add proc stat to slice later
https://github.com/cloudradar-monitoring/cagent/blob/6dcb003285c33e0139d76ebd4d865734024e9cad/processes_notwindows.go#L132

Reply:
ok, but we can still remove the `else` by inverting the logic and having `procs = append(procs, stat)` in before the continue as well right?
I wanna minimise the `if/else` occurrence in the code

Reply:
we can copy append, yes. But later we may add something after it and introduce a bug

https://github.com/cloudradar-monitoring/cagent/blob/f366c916f3dcbd0461dd433581cab698d1181297/processes_notwindows.go#L99-L132

Reply:
we can convert inner-range code into separate func and save one level :)
We can also convert each proc file processing into separate func. But for other funcs this will gives overhead imho

Reply:
ah okay... well then let's keep it as is